Regression with DatePicker and checkboxes on stock page (catalogue > stock)

PierreRambaud opened the pull request:
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.4.x
| Description?  | DatePicker field not working due to wrong css selector. Wrong usage of v-model in PsCheckbox.vue (https://fr.vuejs.org/v2/guide/forms.html#Checkbox)
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Partial http://forge.prestashop.com/browse/BOOM-5620
| How to test?  | Try to click on datepicker field, for example in product page, Available quantity field.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/9133)
<!-- Reviewable:end -->

PierreRambaud commented:
Same problem as @mickaelandrieu on #9137 
@fatmaBouchekoua @khouloudbelguith Can you check please :)

fatmaBouchekoua commented:
Hello @PierreRambaud , 
I'm checking it. That's weird because it works perfectly on local.

PierreRambaud commented:
@fatmaBouchekoua Could you please send me the command you used to run this test? :)

PierreRambaud commented:
@fatmaBouchekoua I can reproduce the bug, while running E2E tests, I see the green message coming and disappeared one second after, and just after the check to `verify the appearance of the green validation`  is executed.

fatmaBouchekoua commented:
Hello @PierreRambaud ,
To run this test I use this command : npm test -- --URL=URLFrontOffice
That's weird, this is working for me on local shop and on Travis https://travis-ci.org/fatmaBouchekoua/PrestaShop/jobs/385647686
I will try to reproduce it on this branch even if this PR doen't have an impact on the product page.

PierreRambaud commented:
@fatmaBouchekoua Hey, I found what happened, a problem with package-lock.json file, it's now fix we will release a patch on 1.7.4.x and develop :)

eternoendless commented on admin-dev/themes/new-theme/js/theme.js:
```diff
@@ -47,6 +42,13 @@ import './product-page/index';
 new Header();
+
+$(() => {
+  initDatePickers();
```
This probably should be done only in pages containing datepickers, shouldn't it?

PierreRambaud replied:
It's a theme feature to have a DatePicker, it's not really specific to a single page.
When this selector isn't found, nothing happened, and it cost maybe 0.0005ms (:trollface:). 
If we decide to move this only in pages containing datepickers, we maybe need to do the same for `product-page/index`, `./clickable-dropdown`, and `./translation-page/index` that are in `theme.js`. Wdyt?

mickaelandrieu replied:
We need to check that adding it globally doesn't conflict with datepickers management in product page.

PierreRambaud replied:
@mickaelandrieu It's a fix because no datepickers working on product page =)

eternoendless replied:
Ok but we will have to do it eventually when we'll split the files per page sometime in the future

eternoendless reviewed (changes requested):
Please commit built assets in a separate commit, it will make rebasing easier

PierreRambaud committed:
Regression with DatePicker and checkboxes on stock page

DatePicker field not working due to wrong css selector.
And because it can be on every page, run datepicker function
directly on theme.js. Fix problem where we can't check checkboxes
on the stock page.

PierreRambaud committed:
Commit assets

PierreRambaud commented:
@eternoendless I rebase everything in two separate commits :) Sorry for the delay

eternoendless commented on admin-dev/themes/new-theme/js/theme.js:
```diff
@@ -47,6 +42,13 @@ import './product-page/index';
 import Header from './header.js';
+import initDatePickers from './app/utils/datepicker';
+
+const $ = global.$;
```
You may need to use `window` instead of `global`

PierreRambaud replied:
Should I need to replace every occurrences  of `global` in other files?

eternoendless replied:
No, it will be automagically converted to `window` by webpack. It's just to avoid linter warnings.

marionf commented:
Hello @PierreRambaud 

It's ok for checkbox
It's ok for datepicker in quantities tab for a standard product

It's not ok for datepicker in combination tab
![capture du 2018-06-04 09-23-17](https://user-images.githubusercontent.com/13449658/40903801-f72f57ac-67d8-11e8-86c6-1081b2595d11.png)

It's not ok for datepicker in stock movement page
![capture du 2018-06-04 09-24-13](https://user-images.githubusercontent.com/13449658/40903839-15c42116-67d9-11e8-8b4b-3603113e67aa.png)

PierreRambaud committed:
Fix datepicker under VueJs

PierreRambaud committed:
assets

PierreRambaud committed:
Wrong selector for datepicker in combinaison page

eternoendless commented:
Review status: all files reviewed at latest revision, all discussions resolved.

---

*[admin-dev/themes/new-theme/js/app/widgets/ps-checkbox.vue, line 49 at r4](https://reviewable.io/reviews/prestashop/prestashop/9133#-LE98YN49owq8CJaSPND:-LE98YN54jL_5HJN51YG:bccm2kn) ([raw file](https://github.com/prestashop/prestashop/blob/5243d55372fd40ad9f99c09e598f7f1379ce608b/admin-dev/themes/new-theme/js/app/widgets/ps-checkbox.vue#L49)):*
> ```Vue
>       onClick() {
>         this.checked = !this.checked;
>       },
> ```

Doesn't removing this prevent the model from being updated?

---


*Comments from [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9133)*
<!-- Sent from Reviewable.io -->

PierreRambaud commented:
Review status: 14 of 15 files reviewed at latest revision, 1 unresolved discussion.

---

*[admin-dev/themes/new-theme/js/app/widgets/ps-checkbox.vue, line 49 at r4](https://reviewable.io/reviews/prestashop/prestashop/9133#-LE98YN49owq8CJaSPND:-LE9D0j_1XxIexuuYk_4:bgyajt8) ([raw file](https://github.com/prestashop/prestashop/blob/5243d55372fd40ad9f99c09e598f7f1379ce608b/admin-dev/themes/new-theme/js/app/widgets/ps-checkbox.vue#L49)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

Doesn't removing this prevent the model from being updated?
</blockquote></details>

No, in fact, model and item aren't used here (only to emit checked event to parent.
And it's against the VueJs philosophy https://fr.vuejs.org/v2/guide/forms.html#Checkbox by setting a v-model on the input, the checked value is already toggled.

---


*Comments from [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9133)*
<!-- Sent from Reviewable.io -->

eternoendless commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

*[admin-dev/themes/new-theme/js/app/widgets/ps-checkbox.vue, line 49 at r4](https://reviewable.io/reviews/prestashop/prestashop/9133#-LE98YN49owq8CJaSPND:-LE9ufLG86VZpiKoTojK:bh25lys) ([raw file](https://github.com/prestashop/prestashop/blob/5243d55372fd40ad9f99c09e598f7f1379ce608b/admin-dev/themes/new-theme/js/app/widgets/ps-checkbox.vue#L49)):*
<details><summary><i>Previously, PierreRambaud (GoT) wrote…</i></summary><blockquote>

No, in fact, model and item aren't used here (only to emit checked event to parent.
And it's against the VueJs philosophy https://fr.vuejs.org/v2/guide/forms.html#Checkbox by setting a v-model on the input, the checked value is already toggled.
</blockquote></details>

![magic.gif](https://files.reviewableusercontent.io/images/reviewable/1009343/Je6XYQ8-5aH26ZZ6Xp8xTvQi/magic.gif)

---


*Comments from [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9133#-:-LE9uw_lCaV-6nv7H_wI:bnfp4nl)*
<!-- Sent from Reviewable.io -->

eternoendless commented:
Thank you @PierreRambaud
